Library/Homebrew/cmd/test-bot.rb

```diff
@@ -637,16 +637,18 @@ def formula(formula_name)
     def homebrew
       @category = __method__
       return if @skip_homebrew
-      test "brew", "tests"
-      if @tap.core_tap?
-        tests_args = ["--no-compat"]
-        readall_args = ["--aliases"]
-        if RUBY_VERSION.split(".").first.to_i >= 2
-          tests_args << "--coverage" if ENV["TRAVIS"]
-          readall_args << "--syntax"
-        end
-        test "brew", "tests", *tests_args
-        test "brew", "readall", *readall_args
+
+      ruby_two = RUBY_VERSION.split(".").first.to_i >= 2
+      tests_args = []
+      tests_args << "--coverage" if ruby_two && ENV["TRAVIS"]
+      test "brew", "tests", *tests_args
+
+      if @tap.nil?
+        test "brew", "tests", "--no-compat"
+      elsif @tap.core_tap?
+        readall_args = []
+        readall_args << "--syntax" if ruby_two
+        test "brew", "readall", "--aliases", *readall_args
```

Comment:
The `--syntax` flag for `brew readall` causes it to check the syntax of Ruby files in `Homebrew/Library/` (package manager files) *in addition* to trying to read all formulae, which I think can be omitted for the taps. On the other I think we want to add `readall --syntax` as a step for the `@tap.nil?` case, otherwise we'll be missing Ruby syntax warnings introduced by PRs in Homebrew/brew.

(A better solution beyond this quick fix might be to untangle `brew readall` a bit, as it performs two very different tasks and there's no way to disable the “read all formulae” function, something we might want to do for Homebrew/brew.)